pkg/depth/buffer.go

```diff
@@ -230,29 +259,43 @@ func (b *Buffer) fetchAndPush() error {
 	}
 
 	var pushUpdates []Update
-	for idx, u := range b.buffer {
-		// skip old events
-		if u.FinalUpdateID <= finalUpdateID {
-			continue
+	if b.isFutures {
```

Comment:
what's the difference between futures and non-futures?

Reply:
For futures, packet loss detection is stricter: it requires ``pu == previous u``. If this condition isn't met, it means data might be missing, so you need to reinitialize the order book.

For spot, as long as ``U <= lastUpdateId <= u``, the order book update continues. But if ``U`` > local order book update ID, you have to rebuild it from scratch.